ui/src/app/guards/access-tasks-guard.service.ts

```diff
@@ -22,21 +22,20 @@ export class AccessTasksGuard implements CanActivate, CanActivateChild {
         route.routeConfig.path.startsWith('nih-callback')) {
       return Observable.from([true]);
     }
-    let enforceRegistered: boolean;
     this.serverConfigService.getConfig().subscribe((config) => {
-      enforceRegistered = config.enforceRegistered;
-    });
+      return this.profileStorageService.profile$.flatMap((profile) => {
+        if ((profile.linkedNihUsername === null || profile.linkedNihUsername === '')
+            && config.enforceRegistered) {
```

Comment:
Seems like you should be checking `enableComplianceLockout` as well here, right?

Reply:
good call :)